File: DomainModel/Cache.cs
﻿using System;

namespace Swk5.GeoCaching.DomainModel {
    public class Cache : IEquatable<Cache> {
        private double cacheDifficulty;
        private double terrainDifficulty;

        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime CreationDate { get; set; }

        public double CacheDifficulty {
            get { return cacheDifficulty; }
            set {
                if (value >= 1 && value <= 5) {
                    cacheDifficulty = value;
                }
                else {
                    throw new ArgumentException();
                }
            }
        }

        public double TerrainDifficulty {
            get { return terrainDifficulty; }
            set {
                if (value >= 1 && value <= 5) {
                    terrainDifficulty = value;
                }
                else {
                    throw new ArgumentException();
                }
            }
        }

        public string Size { get; set; }
        public int OwnerId { get; set; }
        public GeoPosition Position { get; set; }
        public string Description { get; set; }

        public bool Equals(Cache other) {
            if (ReferenceEquals(null, other)) {
                return false;
            }
            if (ReferenceEquals(this, other)) {
                return true;
            }
            return Id == other.Id;
        }

        public override bool Equals(object obj) {
            if (ReferenceEquals(null, obj)) {
                return false;
            }
            if (ReferenceEquals(this, obj)) {
                return true;
            }
            if (obj.GetType() != GetType()) {
                return false;
            }
            return Equals(( Cache ) obj);
        }

        public override string ToString() {
            return string.Format("Id: {0}, Name: {1}, size: {2}", Id, Name, Size);
        }

        public override int GetHashCode() {
            return Id;
        }
    }
}

File: GeoCachingWebClient/src/at/wea5/geocaching/domainmodel/Cache.java
package at.wea5.geocaching.domainmodel;

import java.util.Date;

public class Cache {

//------------------------------------ constructor ------------------------------------
    
    public Cache(double cacheDifficulty, double terrainDifficulty, int id, String name, Date creationDate, String size, int ownerId,
            GeoPosition position, String description) {
        
        this.cacheDifficulty = cacheDifficulty;
        this.terrainDifficulty = terrainDifficulty;
        this.id = id;
        this.name = name;
        this.creationDate = creationDate;
        Size = size;
        this.ownerId = ownerId;
        this.position = position;
        this.description = description;
    }

//-------------------------------------- public ---------------------------------------
    
    public double getCacheDifficulty() {
        return cacheDifficulty;
    }
    
    public void setCacheDifficulty(double cacheDifficulty) {
        this.cacheDifficulty = cacheDifficulty;
    }
    
    public double getTerrainDifficulty() {
        return terrainDifficulty;
    }
    
    public void setTerrainDifficulty(double terrainDifficulty) {
        this.terrainDifficulty = terrainDifficulty;
    }
    
    public int getId() {
        return id;
    }
    
    public void setId(int id) {
        this.id = id;
    }
    
    public String getName() {
        return name;
    }
    
    public void setName(String name) {
        this.name = name;
    }
    
    public Date getCreationDate() {
        return creationDate;
    }
    
    public void setCreationDate(Date creationDate) {
        this.creationDate = creationDate;
    }
    
    public String getSize() {
        return Size;
    }
    
    public void setSize(String size) {
        Size = size;
    }
    
    public int getOwnerId() {
        return ownerId;
    }
    
    public void setOwnerId(int ownerId) {
        this.ownerId = ownerId;
    }
    
    public GeoPosition getPosition() {
        return position;
    }
    
    public void setPosition(GeoPosition position) {
        this.position = position;
    }
    
    public String getDescription() {
        return description;
    }
    
    public void setDescription(String description) {
        this.description = description;
    }

//------------------------------------- private ---------------------------------------

//-------------------------------------- members --------------------------------------
    private double cacheDifficulty;
    private double terrainDifficulty;

    private int id;
    private String name;
    private Date creationDate;

    private String Size;
    private int ownerId;
    private GeoPosition position;
    private String description;

}
